src/suites/cts/validation/create_bind_group_layout.spec.ts

```diff
@@ -0,0 +1,87 @@
+export const description = `
+Create bind group layout validation tests.
+`;
+
+import { TestGroup } from '../../../framework/index.js';
+
+import { ValidationTest } from './validation_test.js';
+
+function clone(descriptor: GPUBindGroupLayoutDescriptor): GPUBindGroupLayoutDescriptor {
+  return JSON.parse(JSON.stringify(descriptor));
+}
+
+export const g = new TestGroup(ValidationTest);
+
+g.test('some binding index was specified more than once', async t => {
+  const goodDescriptor: GPUBindGroupLayoutDescriptor = {
+    bindings: [
+      {
+        binding: 0,
+        visibility: GPUShaderStage.COMPUTE,
+        type: 'storage-buffer',
+      },
+      {
+        binding: 1,
+        visibility: GPUShaderStage.COMPUTE,
+        type: 'storage-buffer',
+      },
+    ],
+  };
+
+  const badDescriptor = clone(goodDescriptor);
+  badDescriptor.bindings![1].binding = 0;
```

Comment:
What's that ! for?

Reply:
it overrides the type of a nullable thing to be not nullable:
https://stackoverflow.com/q/42273853